src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1065,6 +1085,199 @@ def _validate_up_args(cmd, source, artifact, build_env_vars, image, repo, regist
+            if config_class_name is not None and config_class_name != "":
+                model_load_class.append(f"AZURE_ML_CONFIG_CLASS_NAME={config_class_name}")
+            task_name = safe_get(model_detail_data, "flavors", "hftransformersv2", "task_type")
+            if task_name is not None and task_name != "":
+                model_load_class.append(f"AZURE_ML_PIPELINE_TASK_NAME={task_name}")
+        elif safe_get(model_detail_data, "flavors", "transformers"):
+            pipeline_tokenizer_type = safe_get(model_detail_data, "flavors", "transformers", "tokenizer_type")
+            pipeline_instance_type = safe_get(model_detail_data, "flavors", "transformers", "instance_type")
+            pipeline_model_type = safe_get(model_detail_data, "flavors", "transformers", "pipeline_model_type")
+            pipeline_task_name = safe_get(model_detail_data, "flavors", "transformers", "task")
+            model_load_class = [f"AZURE_ML_PIPELINE_INSTANCE_TYPE={pipeline_instance_type}",
+                                f"AZURE_ML_PRETRAINED_MODEL_CLASS_NAME={pipeline_model_type}",
+                                f"AZURE_ML_PIPELINE_TASK_NAME={pipeline_task_name}",
+                                f"AZURE_ML_TOKENIZER_CLASS_NAME={pipeline_tokenizer_type}"]
+            if pipeline_tokenizer_type is None or pipeline_instance_type is None or pipeline_model_type is None or pipeline_task_name is None:
+                logger.warning("Model loading class names cannot be automatically detected.")
+                model_load_class = []
+    elif model_format.lower() == "custom":
+        raise ValidationError("Custom model format is not supported yet.")
+    return model_asset_id, model_reference_endpoint, model_format, model_load_class
+
+
+def _validate_azml_args(cmd, default_mcr_img, image_name, model_registry, model_name, model_version):
+    ACA_AZML_BLESSED_MODEL = ["azureml:phi-4",
+                              "azureml:mistralai-mistral-7b-v01",
+                              "azureml:phi-3.5-mini-instruct",
+                              "azureml:gpt2-medium",
+                              "azureml:phi-4-mini-reasoning",
+                              "azureml:phi-4-reasoning"]
+    if model_registry is None and model_name is None and model_version is None:
+        return
+    if model_registry is None:
+        raise RequiredArgumentMissingError(
+            "You must specify --model-registry when deploying Foundry Models through Azure Container Apps CLI."
+        )
+    if model_name is None:
+        raise RequiredArgumentMissingError(
+            "You must specify --model-name when deploying Foundry Models through Azure Container Apps CLI."
+        )
+    if model_version is None:
+        raise RequiredArgumentMissingError(
+            "You must specify --model-version when deploying Foundry Models through Azure Container Apps CLI."
+        )
+    elif image_name is None or image_name.lower() == default_mcr_img:
```

Comment:
Style error:

src/containerapp/azext_containerapp/_up_utils.py:1177:4: R1720: Unnecessary "elif" after "raise", remove the leading "el" from "elif" (no-else-raise)